portal/src/components/Materials/MaterialSet/MaterialSet.vue

```diff
@@ -38,25 +38,12 @@
 export default {
   name: 'MaterialSet',
   props: {
-    material: {
-      type: Object,
-      default: null,
-      required: false
+    setMaterials: {
+      type: Array,
+      default: () => [],
+      required: true
     }
   },
-  data() {
-    return {
-      setMaterials: []
-    }
-  },
-  watch: {
-    material: function() {
-      this.updateSetMaterials()
```

Comment:
This call is now missing in the parent component. I think the material can change and then this watch callback triggers. At that point we would need to do the fetchSetMaterials as well. If I remember correctly it happens when you navigate? Or when you hard refresh on the page itself? I wouldn't mind if the material didn't update of course and that we didn't have to do a re-fetch, but adding this seemed a quick fix at the time and my guess is we still need it.

Reply:
I added a key with the material external id to the MaterialInfo component, so instead of updating the component it mounts a new one when there is another material because the key changes. This way it will fetch the new set materials when you navigate to another material.

Reply:
Ok, nice! Much better solution. Cheers :+1: